server

```diff
@@ -63,8 +66,24 @@ end
 
 post '/boards/:board/cards' do
   board = boards[params[:board]]
+  card_id = SecureRandom.uuid
   redirect '/' unless board
 
-  cards << params.slice(:board, :column, :message).merge(username: cookies[:username])
+  card = params.slice(:board, :column, :message).merge(username: cookies[:username]).merge(votes: 0).merge(id: card_id)
+  card = JSON.parse(card.to_json, symbolize_names: true)
+  cards << card
+
+  redirect "/boards/#{params[:board]}"
+end
+
+get '/boards/:board/cards/:card/up_vote' do
+  board = params[:board]
+  card_id = params[:card]
+  card_index = cards.find_index{|card| card[:id] == card_id }
+  redirect '/' unless card_id
+  redirect "/boards/#{params[:board]}" if cookies[:rem_votes].to_i == 0
+  cards[card_index][:votes] += 1
+  cookies[:rem_votes] = cookies[:rem_votes].to_i - 1
+
   redirect "/boards/#{params[:board]}"
```

Comment:
What's missing here is we need to have the votes an array of users identifiers that voted, so we can then calculate the total votes for this user and stop him from voting more than the max of the board
```suggestion
  card = cards.find {|card| card[:id] == params[:card] }
  redirect "/boards/#{board[:id]}/cards" unless card
  
  card[:votes] += 1 
  redirect "/boards/#{board[:id]}/cards"
```